pilot/pkg/serviceregistry/kube/controller/ambient/workloads.go

```diff
@@ -253,7 +279,8 @@ func (a *index) podWorkloadBuilder(
 			}))
 		}
 		services := krt.Fetch(ctx, WorkloadServices, fo...)
-		// Logic from https://github.com/howardjohn/kubernetes/blob/7c873327b679a70337288da62b96dd610858181d/staging/src/k8s.io/endpointslice/utils.go#L37
+		services = append(services, a.matchingServicesWithoutSelectors(ctx, p, services, WorkloadServices, EndpointSlices, EndpointSlicesAddressIndex)...)
```

Comment:
Isn't this too bad performance, each time a pod event will trigger selecting all endpointslice

Reply:
It's not selecting all of them, there is an efficient O(1) lookup into an index.

Reply:
Oh, an index is there